File: two-pointers/two-sum-ii.go
package main

import "fmt"

func TwoSumDemo() {
	// numbers := []int{2, 7, 11, 15}
	numbers := []int{2, 3, 4}
	target := 6
	result := TwoSum(numbers, target)
	fmt.Println(result)
}

func TwoSum(numbers []int, target int) []int {
	arr := map[int]int{}
	for i, v := range numbers {
		need := target - v
		if value, err := arr[need]; err {
			return []int{value + 1, i + 1}
		}
		arr[v] = i
	}
	return nil
}

File: two-pointers/two-sum-ii.py
class Solution:
  @staticmethod
  def twoSum(numbers: list[int], target: int) -> list[int]:
    left = 0
    right = len(numbers) -1
    while left < right:
      sum = numbers[left] + numbers[right]
      if sum > target:
        right -= 1
      elif sum < target:
        left += 1
      else:
        return [left + 1, right + 1]